crates/node/engine/src/task_queue/tasks/build/error.rs

```diff
@@ -64,13 +75,26 @@ impl EngineTaskError for BuildTaskError {
         match self {
             Self::ForkchoiceUpdateFailed(inner) => inner.severity(),
             Self::PayloadInsertionFailed(inner) => inner.severity(),
-            Self::NoForkchoiceUpdateNeeded => EngineTaskErrorSeverity::Temporary,
-            Self::EngineSyncing => EngineTaskErrorSeverity::Temporary,
+            Self::EngineBuildError(EngineBuildError::FinalizedAheadOfUnsafe(_, _)) => {
+                EngineTaskErrorSeverity::Critical
+            }
+            Self::EngineBuildError(EngineBuildError::AttributesInsertionFailed(_)) => {
+                EngineTaskErrorSeverity::Temporary
+            }
+            Self::EngineBuildError(EngineBuildError::InvalidPayload(_)) => {
+                EngineTaskErrorSeverity::Temporary
+            }
+            Self::EngineBuildError(EngineBuildError::UnexpectedPayloadStatus(_)) => {
+                EngineTaskErrorSeverity::Temporary
+            }
```

Comment:
To check: I think that this should instead be
```suggestion
Self::EngineBuildError(EngineBuildError::UnexpectedPayloadStatus(_)) => {
                EngineTaskErrorSeverity::Critical
            }
```
The `forkchoice_updated` should *never* return the Accepted variant